clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -656,8 +750,30 @@ void
 offline_instru_t::bb_analysis(void *drcontext, void *tag, void **bb_field,
                               instrlist_t *ilist, bool repstr_expanded)
 {
-    *bb_field = (void *)(ptr_uint_t)instru_t::count_app_instrs(ilist);
+    per_block_t *per_block =
+        reinterpret_cast<per_block_t *>(dr_thread_alloc(drcontext, sizeof(*per_block)));
+    *bb_field = per_block;
+
+    per_block->instr_count = instru_t::count_app_instrs(ilist);
+
     identify_elidable_addresses(drcontext, ilist, OFFLINE_FILE_VERSION);
+
+    app_pc tag_pc = dr_fragment_app_pc(tag);
+    if (drmodtrack_lookup(drcontext, tag_pc, nullptr, nullptr) != DRCOVLIB_SUCCESS) {
+        // For (unmodified) library code we do not need to record encodings as we
```

Comment:
Yes, it's mostly in the final trace file where we want to include encodings for all code, b/c that's further separated from the tracing and the binaries in space (use case of analyzing a Windows trace on a Linux machine where it's a pain to examine Windows binaries) and time (hard to keep the binaries around forever).  For the raw we assume it will be post-processed quickly and it's worth the overhead reduction of requiring the binaries there.